### PR TITLE
 Add navigation controller top view controller as fallback

### PR DESCRIPTION
**Pull request by zenangst**
If the root view controller is a navigation controller, it will now use the top view controller (which is the view controller at the top of the navigation stack) as a fallback if no previous nib names have been found on the previous match.

### CODE REVIEW

File: InjectionBundle/SwiftEval.swift [hunk 1]
```diff
@@ -220,7 +220,7 @@ public class SwiftEval: NSObject {
                         if ($line =~ /^\\s*cd /) {
                             $realPath = $line;
                         }
-                        elsif (my ($product) = $line =~ m@\\.xcent --timestamp=none (.*)\\r@o) {
+                        elsif (my ($product) = $line =~ m@\\.xcent --timestamp=none (.*\\.app)\\r@o) {
```
Comment (zenangst) [inline InjectionBundle/SwiftEval.swift]: Curious to why this shows up, this should already be on `master`

File: InjectionBundle/InjectionClient.mm [hunk 2]
```diff
@@ -219,17 +219,23 @@ - (BOOL)injectUI:(NSString *)changed {
...
             visibleVC.childViewControllers.firstObject)
             visibleVC = child;
         if ([visibleVC respondsToSelector:@selector(viewControllers)])
             visibleVC = [(UISplitViewController *)visibleVC
                          viewControllers].lastObject;
+
         if ([visibleVC respondsToSelector:@selector(visibleViewController)])
             visibleVC = [(UINavigationController *)visibleVC
                          visibleViewController];
+        if (!visibleVC.nibName && navigationController) {
```
Comment (johnno1962) [inline InjectionBundle/InjectionClient.mm]: Could this be !visibleVC.nibName && [navigationController respondsToSelector:@selector(topViewController)]?

File: InjectionBundle/SwiftEval.swift (hunk 1 above)
Comment (johnno1962) [reply InjectionBundle/SwiftEval.swift]: It’ll all come out in the wash but as you merge can you be sure to Squash?

File: InjectionBundle/InjectionClient.mm (hunk 2 above)
Comment (zenangst) [reply InjectionBundle/InjectionClient.mm]: Fixed! :)

**Comment by johnno1962** [comment]
No squash?

**Comment by johnno1962** [comment]
I’ve tweaked the history and done a forced update of master. All good now.

**Comment by zenangst** [comment]
Yeah you can do that ;)

**Pull request merged**